.github/workflows/automation-scripts/update-changelogs.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const path = require('path');
+
+// Read all files in the directory
+function getFilePaths(directoryPath) {
+  let filePaths = [];
+  let files = fs.readdirSync(directoryPath);
+  files.forEach(file => {
+    let filePath = path.join(directoryPath, file);
+    filePaths.push(filePath);
+  });
+  return filePaths
+}
+
+function loadJsonFiles(filePath) {
+  // Load each JSON file
+  data = fs.readFileSync(filePath);
+  const jsonData = JSON.parse(data);
+  return jsonData;
+}
+
+// returns true if currentVersion is greater than incoming
+function compareVersions(currentVersion, incomingVersion) {
+  currentVersion = currentVersion.split('.').map(Number);
+  incomingVersion = incomingVersion.split('.').map(Number);
+  const currentDif = currentVersion.map((num, index) => {
+    return num - incomingVersion[index];
+  })
+  const leadNum = currentDif.filter((num) => { return num !== 0 })[0];
+  return leadNum > 0;
+}
+
+function findNewEntries(currentEntries, incomingEntries) {
+  currVersions = [];
+  incomingVersions = [];
+
+  currentEntries.forEach((entry) => {
+    currVersions.push(entry.version);
+  });
+
+  incomingEntries.forEach((entry) => {
+    incomingVersions.push(entry.version);
+  });
+
+  newEntries = [];
+  incomingVersions.forEach((version, i) => {
+    if (!currVersions.includes(version))
+      newEntries.push(incomingEntries[i]);
+  })
+
+  return newEntries;
+}
+
+function addEntriesToCurr(currentEntries, newEntries) {
+  let i = 0;
+  let j = 0;
+  while (j !== newEntries.length) {
+    if (!compareVersions(currentEntries[i].version, newEntries[j].version)) {
+      currentEntries.splice(i, 0, newEntries[j])
+      j++;
+    }
+    i++;
+  }
+  return currentEntries;
+}
+
+function fixChangeLogs(currentFiles, incomingFiles) {
```

Comment:
feel like algorithmically there is an easier way to do this

you have 2 arrays, and you basically want the union of them, where the union contains only unique values and is sorted

Reply:
I didn't see a union functionality from JS except for in some other libraries, I don't know JS all that well so i could be wrong. I kind of just implemented my own, and I agree its not very simple, but its fairly efficient, at least compared to the other simpler ways I could think of writing it.

Reply:
Perhaps a merge and then sort? [...currentEntries, ...newEntries].sort(customSortFunction)

Reply:
Ah I see they're objects not version strings...

Reply:
yeah, I was just avoiding a custom sort function as I didn't know how to implement it off the top of my head, and figured I could write this faster. I'll try this and get something as simple as possible

Reply:
Used this method, and tested with the new change logs for 3.7.12, looks good. Still going to clean up some of the syntax, and try to condense into one file